cmd/config.go

```diff
@@ -21,44 +21,75 @@ package cmd
 
 import (
 	// 'go lint': need blank import for embedding default config
+	"bytes"
+	// needed for embedded filesystem
 	_ "embed"
+	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 
+	"github.com/facebookincubator/ttpforge/pkg/logging"
+	"github.com/facebookincubator/ttpforge/pkg/repos"
 	"github.com/spf13/afero"
+	"gopkg.in/yaml.v3"
 )
 
+// Config stores the variables from the TTPForge global config file
+type Config struct {
+	RepoSpecs []repos.Spec `yaml:"repos"`
+
+	repoCollection repos.RepoCollection
+	cfgFile        string
+}
+
 var (
-	autoInitConfig bool
 	//go:embed default-config.yaml
 	defaultConfigContents string
 	defaultConfigFileName = "config.yaml"
 	defaultResourceDir    = ".ttpforge"
+
+	// Conf refers to the configuration used throughout TTPForge.
+	Conf = &Config{}
+
+	logConfig logging.Config
 )
 
-func ensureDefaultConfig() (string, error) {
+func getDefaultConfigFilePath() (string, error) {
 	homeDir, err := os.UserHomeDir()
 	if err != nil {
 		return "", err
 	}
+	defaultConfigPath := filepath.Join(homeDir, defaultResourceDir, defaultConfigFileName)
+	return defaultConfigPath, nil
+}
 
-	f := afero.NewOsFs()
-	defaultConfigDir := filepath.Join(homeDir, defaultResourceDir)
-	defaultConfigPath := filepath.Join(defaultConfigDir, defaultConfigFileName)
-	exists, err := afero.Exists(f, defaultConfigPath)
+// loadRepoCollection verifies that all repositories specified
+// in the configuration file are present on the filesystem
+// and clones missing ones if needed
+func (cfg *Config) loadRepoCollection() (repos.RepoCollection, error) {
+	// locate our config file directory to expend config-relative paths
+	cfgFileAbsPath, err := filepath.Abs(cfg.cfgFile)
 	if err != nil {
-		return "", err
-	}
-	if exists {
-		return defaultConfigPath, nil
+		return nil, err
 	}
+	cfgDir := filepath.Dir(cfgFileAbsPath)
+	fsys := afero.NewOsFs()
 
-	if err = os.MkdirAll(defaultConfigDir, 0700); err != nil {
-		return "", err
-	}
+	return repos.NewRepoCollection(fsys, cfg.RepoSpecs, cfgDir)
+}
 
-	if err := afero.WriteFile(f, defaultConfigPath, []byte(defaultConfigContents), 0600); err != nil {
-		return "", err
+// save() writes the current config back to its file - used by `install“ command
+func (cfg *Config) save() error {
```

Comment:
this new function (used to save config after repo changes) is the key change for this file - everything else is just cut-and-paste to organize a bit